lib/tizen_project.dart

```diff
@@ -21,15 +26,39 @@ class TizenProject extends FlutterProjectPlatform {
 
   final FlutterProject parent;
 
+  /// See: [FlutterManifest] in `flutter_manifest.xml`
+  late final Map<String, Object?> _parentPubspec = () {
+    final File manifestFile = parent.directory.childFile(defaultManifestPath);
+    if (!manifestFile.existsSync()) {
+      return <String, Object?>{};
+    }
+    YamlMap? yamlMap;
+    try {
+      yamlMap = loadYaml(manifestFile.readAsStringSync()) as YamlMap?;
+    } on YamlException {
+      return <String, Object?>{};
+    }
+    return yamlMap?.cast<String, Object?>() ?? <String, Object?>{};
+  }();
+
   @override
   String get pluginConfigKey => TizenPlugin.kConfigKey;
 
   Directory get editableDirectory => parent.directory.childDirectory('tizen');
+  Directory get _ephemeralModuleDirectory =>
+      parent.directory.childDirectory('.tizen');
+
+  Directory get hostAppRoot {
+    if (!parent.isModule || editableDirectory.existsSync()) {
+      return isMultiApp ? uiAppDirectory : editableDirectory;
+    }
+    return _ephemeralModuleDirectory;
+  }
```

Comment:
`hostAppRoot` may refer to either
- `.tizen/` (ephemeralModuleDirectory)
- `tizen/` (editableDirectory)
- `tizen/ui/` (uiAppDirectory)

according to the context.